commit d25141a818383b3c3b09f065698c544a7a0ec6e7
Author: Sukadev Bhattiprolu <sukadev@linux.vnet.ibm.com>
Date:   Wed Oct 29 14:01:11 2008 -0700

    'kill sig -1' must only apply to caller's namespace
    
    Currently "kill <sig> -1" kills processes in all namespaces and breaks the
    isolation of namespaces.  Earlier attempt to fix this was discussed at:
    
    	http://lkml.org/lkml/2008/7/23/148
    
    As suggested by Oleg Nesterov in that thread, use "task_pid_vnr() > 1"
    check since task_pid_vnr() returns 0 if process is outside the caller's
    namespace.
    
    Signed-off-by: Sukadev Bhattiprolu <sukadev@linux.vnet.ibm.com>
    Acked-by: Eric W. Biederman <ebiederm@xmission.com>
    Tested-by: Daniel Hokka Zakrisson <daniel@hozac.com>
    Signed-off-by: Oleg Nesterov <oleg@redhat.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Adjusted to apply to Debian's 2.6.24 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.24.orig/kernel/signal.c linux-source-2.6.24/kernel/signal.c
--- linux-source-2.6.24.orig/kernel/signal.c	2009-04-11 14:35:50.000000000 -0600
+++ linux-source-2.6.24/kernel/signal.c	2009-04-18 14:52:22.000000000 -0600
@@ -1150,7 +1150,8 @@ static int kill_something_info(int sig, 
 
 		read_lock(&tasklist_lock);
 		for_each_process(p) {
-			if (p->pid > 1 && !same_thread_group(p, current)) {
+			if (task_pid_vnr(p) > 1 &&
+					!same_thread_group(p, current)) {
 				int err = group_send_sig_info(sig, info, p);
 				++count;
 				if (err != -EPERM)
